pyiode/pyiode_model.pyx

```diff
@@ -6,45 +6,246 @@
+                        pre_listname: str = "_PRE", 
+                        inter_listname: str = "_INTER", 
+                        post_listname: str = "_POST",
+                        eps: float = 0.0001, 
+                        relax: float = 1.0, 
+                        maxit: int = 100, 
+                        init_values: int = KV_INIT_TM1, 
+                        debug: int = 0, 
+                        newton_eps: float = 1e-6, 
+                        newton_maxit: int = 50, 
+                        newton_debug: int = 0):
+    '''
+    Simulate a model defined by the 3 Connex Components (stored in 3 lists).
+    '''
+
+    model_simulate_save_parms(eps, relax, maxit, init_values, -1, -1)
+    
+    if IodeModelSimulateSCC(cstr(sample_from), cstr(sample_to), 
+                         cstr(pre_listname), cstr(inter_listname), cstr(post_listname),
+                         eps, relax, maxit, init_values, debug, 
+                         newton_eps, newton_maxit, newton_debug):
+        raise RuntimeError(f"model_simulate_scc() failed")
+        
+
+def model_simulate_save_parms(
+                    eps: float = 0.0001, 
+                    relax: float = 1.0, 
+                    maxit: int = 100, 
+                    init_values: int = KV_INIT_TM1, 
+                    sort_algo: int = SORT_BOTH, 
+                    nb_passes: int = 1):
+    ''' 
+        Save in the global variabels KSIM_* the simulation parameters used during the last 
+        call to model_simulate() and model_simulate_scc(). 
+        The purpose of this function is to enable later reporting
+        via the functions model_simulate_maxit(), model_simulate_eps()...
+    '''
+    global KSIM_MAXIT
+    global KSIM_EPS
+    global KSIM_RELAX
+    global KSIM_PASSES
+    global KSIM_SORT
+    global KSIM_START
+
+    KSIM_MAXIT = maxit
+    KSIM_EPS = eps
+    KSIM_RELAX = relax
+    KSIM_START = init_values
+
+    if nb_passes >= 0: KSIM_PASSES = nb_passes  # not used by model_simulate_scc()
+    if sort_algo >= 0: KSIM_SORT = sort_algo    # id.
+    
+   
+def model_simulate_maxit() -> int:
+    ''' Returns the maxit parameter of the last simulation '''
+    return KSIM_MAXIT
+
+def model_simulate_eps() -> float:
+    ''' Returns the eps parameter of the last simulation '''
+    return KSIM_EPS
+
+def model_simulate_relax() -> float:
+    ''' Returns the relax parameter of the last simulation '''
+    return KSIM_RELAX
+
+def model_simulate_nb_passes() -> int:
+    ''' Returns the nb_passes parameter of the last simulation '''
+    return KSIM_PASSES
+
+def model_simulate_sort_algo() -> int:
+    ''' 
+    Returns the sort_algo parameter of the last simulation: 
+        - iode.SORT_CONNEX (0) for SCC decomposition only
+        - iode.SORT_BOTH (1)   for SCC decomposition and triangulation
+        - iode.SORT_NONE (2)   for no reordering nor SCC decomposition
+    '''
+    
+    return KSIM_SORT
```

Comment:
return __KSIM_SORT

Reply:
idem

Reply:
If KSIM_SORT is declared in a `.pxd` file and imported via `cimport`, you can put this conversation as resolved